resources/assets/js/components/JobDetails/JobDetailsPage.tsx

```diff
@@ -0,0 +1,145 @@
+import { connect } from "react-redux";
+import RootContainer from "../RootContainer";
+import ProgressTracker from "../ProgressTracker/ProgressTracker";
+import { ProgressTrackerItem } from "../ProgressTracker/types";
+
+interface JobDetailsPageProps {
+  jobId: number | null;
+  job: Job | null;
+  loadJob: (jobId: number) => void;
+  handleCreateJob: (newJob: Job) => Promise<boolean>;
+  handleUpdateJob: (newJob: Job) => Promise<boolean>;
+}
+
+const JobDetailsPage: React.FunctionComponent<JobDetailsPageProps> = ({
+  jobId,
+  job,
+  loadJob,
+  handleCreateJob,
+  handleUpdateJob,
+}): React.ReactElement => {
+  useEffect((): void => {
+    if (jobId) {
+      loadJob(jobId);
+    }
+  }, [jobId, loadJob]);
+
+  const waitingForJob = jobId !== null && job === null;
+  const handleModalCancel = (): void => {};
+  const handleModalConfirm = (): void => {
+    window.location.href = "/manager/jobs";
+  }; // TODO: go to next page
+  const handleSubmit = job ? handleUpdateJob : handleCreateJob;
+  const progressTrackerItems: ProgressTrackerItem[] = [
+    { state: "active", label: "Step 01", title: "Job Info" },
+    { state: "complete", label: "Step 02", title: "Work Env." },
+    { state: "error", label: "Step 03", title: "Impact" },
+  ];
+  return (
+    <section>
+      <ProgressTracker
+        items={progressTrackerItems}
+        backgroundColor="black"
+        fontColor="white"
+        classNames="manager-jpb-tracker"
+        itemsWrapperClassNames="tracker manager-jpb-tracker-wrapper"
+      />
+      {waitingForJob ? (
+        <div
+          data-c-container="form"
+          data-c-padding="top(triple) bottom(triple)"
+        >
+          <h3
+            data-c-font-size="h3"
+            data-c-font-weight="bold"
+            data-c-margin="bottom(double)"
+          >
+            Job Loading...
+          </h3>
+        </div>
+      ) : (
+        <JobDetailsIntl
+          job={job}
+          handleSubmit={handleSubmit}
+          handleModalCancel={handleModalCancel}
+          handleModalConfirm={handleModalConfirm}
+        />
+      )}
+    </section>
+  );
+};
+
+const mapStateToPropsPage = (
+  state: RootState,
+): {
+  job: Job | null;
+  getJob: (jobId: number | null) => Job | null;
+} => ({
+  job: getSelectedJob(state),
+  getJob: (jobId: number | null): Job | null =>
+    jobId ? selectJob(state, { jobId }) : null,
+});
+
+const mapDispatchToPropsPage = (
+  dispatch: DispatchType,
+): {
+  loadJob: (jobId: number) => void;
+  handleCreateJob: (newJob: Job) => Promise<boolean>;
+  handleUpdateJob: (newJob: Job) => Promise<boolean>;
+} => ({
+  loadJob: (jobId: number): void => {
+    dispatch(fetchJob(jobId));
+    dispatch(setSelectedJob(jobId));
+  },
+  handleCreateJob: async (newJob: Job): Promise<boolean> => {
+    const result = await dispatch(createJob(newJob));
+    if (!result.error) {
+      console.log("Handling create job result");
+      // result.payload.then(resultJob => {
```

Comment:
Is this supposed to be commented?

Reply:
Its supposed to be deleted, I did the same thing with await instead.